cmd/controller-manager/app/controllermanager.go

```diff
@@ -546,12 +546,13 @@ func setupControllers(mgr controllerruntime.Manager, opts *options.Options, stop
 
 	if features.FeatureGate.Enabled(features.PropagateDeps) {
 		dependenciesDistributor := &dependenciesdistributor.DependenciesDistributor{
-			Client:              mgr.GetClient(),
-			DynamicClient:       dynamicClientSet,
-			InformerManager:     controlPlaneInformerManager,
-			ResourceInterpreter: resourceInterpreter,
-			RESTMapper:          mgr.GetRESTMapper(),
-			EventRecorder:       mgr.GetEventRecorderFor("dependencies-distributor"),
+			Client:                          mgr.GetClient(),
+			DynamicClient:                   dynamicClientSet,
+			InformerManager:                 controlPlaneInformerManager,
+			ResourceInterpreter:             resourceInterpreter,
+			RESTMapper:                      mgr.GetRESTMapper(),
+			EventRecorder:                   mgr.GetEventRecorderFor("dependencies-distributor"),
+			ConcurrentResourceTemplateSyncs: opts.ConcurrentResourceTemplateSyncs,
```

Comment:
The option `--concurrent-resource-template-syncs` is kind of dedicated to `resource detector`, its value can be large and beyond what the dependency distributor actually requires.

My suggestion reverting this change from this PR, let the PR focus on distinguishing the queue metrics. And if you want to configure the concurrent value of `distributor`, then we add a new flag for it.